modules.d/90crypt/module-setup.sh

```diff
@@ -92,6 +92,73 @@ cmdline() {
 # called by dracut
 install() {
 
+    # systemd only - check if the key file is a socket and if so, include required units
+    install_socket_units() {
+        local volume_name=$1
+        local socket_path=$2
+
+        # ignore paths followed by a device specification
+        if [[ "$socket_path" == *":"* ]]; then
+            return
+        fi
+
+        # if no explicit path is provided, try to include units for auto-discoverable keys
+        if [[ "$socket_path" = "-" || "$socket_path" = "none" ]]; then
```

Comment:
Can you use `==` here too? Also please split this into `[[ ... ]] || [[ ... ]]`

Reply:
Done. Also dropped the quotes around `"$socket_path"`, as linter also complained about this.